src/Views/Text.vala

```diff
@@ -30,20 +28,22 @@ public class PantheonShell.Text : Gtk.Grid {
     private const string OD_DOC_FONT = "OpenDyslexic Regular 10";
     private const string OD_MON_FONT = "OpenDyslexicMono Regular 10";
 
-    private const double[] TEXT_SCALE = {0.75, 1, 1.25, 1.5};
-
-    private Granite.Widgets.ModeButton text_size_modebutton;
-
     construct {
         var text_size_label = new Gtk.Label (_("Size:")) {
-            halign = Gtk.Align.END
+            halign = Gtk.Align.END,
+            valign = Gtk.Align.START
         };
 
-        text_size_modebutton = new Granite.Widgets.ModeButton ();
-        text_size_modebutton.append_text (_("Small"));
-        text_size_modebutton.append_text (_("Default"));
-        text_size_modebutton.append_text (_("Large"));
-        text_size_modebutton.append_text (_("Larger"));
+        var text_size_adjustment = new Gtk.Adjustment (-1, 0.75, 1.5, 0.05, 0.05, 0.05);
+
+        var text_size_scale = new Gtk.Scale (Gtk.Orientation.HORIZONTAL, text_size_adjustment) {
+            draw_value = false
+        };
+        text_size_scale.add_mark (0.75, Gtk.PositionType.BOTTOM, _("Small"));
+        text_size_scale.add_mark (1, Gtk.PositionType.BOTTOM, _("Default"));
+        text_size_scale.add_mark (1.25, Gtk.PositionType.BOTTOM, null);
+        // Avoid rendering bug when putting mark at the very end
+        text_size_scale.add_mark (1.45, Gtk.PositionType.BOTTOM, _("Largest"));
```

Comment:
This makes me wonder if we need the small and largest marks at all; they're kind of self-evident based on the design, and we would avoid this funkiness.